operator/builtin/parser/xml/xml.go

```diff
@@ -0,0 +1,110 @@
+package xml
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+)
+
+func init() {
+	operator.Register("xml_parser", func() operator.Builder { return NewXMLParserConfig("") })
+}
+
+// NewXMLParserConfig creates a new XML parser config with default values
+func NewXMLParserConfig(operatorID string) *XMLParserConfig {
+	return &XMLParserConfig{
+		ParserConfig: helper.NewParserConfig(operatorID, "xml_parser"),
+	}
+}
+
+// XMLParserConfig is the configuration of an XML parser operator.
+type XMLParserConfig struct {
+	helper.ParserConfig `yaml:",inline"`
+}
+
+// Build will build an XML parser operator.
+func (c XMLParserConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	parserOperator, err := c.ParserConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	xmlParser := &XMLParser{
+		ParserOperator: parserOperator,
+	}
+
+	return []operator.Operator{xmlParser}, nil
+}
+
+// XMLParser is an operator that parses XML.
+type XMLParser struct {
+	helper.ParserOperator
+}
+
+// Process will parse an entry for XML.
+func (x *XMLParser) Process(ctx context.Context, entry *entry.Entry) error {
+	return x.ParserOperator.ProcessWith(ctx, entry, parse)
+}
+
+// parse will parse an xml value
+func parse(value interface{}) (interface{}, error) {
+	strValue, ok := value.(string)
+	if !ok {
+		return nil, fmt.Errorf("value is not a string")
+	}
+
+	reader := strings.NewReader(strValue)
+	decoder := xml.NewDecoder(reader)
```

Comment:
Should we consider allowing a user to configure the Decoder to *not* be Strict? (See source for examples of what !Strict allows: https://golang.org/src/encoding/xml/xml.go )

Reply:
Good catch. I think it doesn't hurt to expose this as a configuration parameter and then default to true.

Reply:
We agreed offline to create another ticket for this feature enhancement.